service/cert-reloader.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2023 LY Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"crypto/tls"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/kpango/glg"
+	"github.com/pkg/errors"
+)
+
+var defaultPollInterval = 1 * time.Minute
+
+// LogFn allows customized logging.
+type LogFn func(format string, args ...interface{})
+
+// CertReloader reloads the (key, cert) pair from the filesystem when
+// the cert file is updated.
+type CertReloader struct {
```

Comment:
Let's separate this as independent library in `garm v4`. Prefer not update to much from the [SIA implementation](https://github.com/AthenZ/k8s-athenz-sia/blob/main/pkg/util/cert-reloader.go).

Reply:
OK

Reply:
Handled in https://github.com/AthenZ/garm/pull/50/commits/2f224862ef8971af729aab619335f1f0b81c407f